build-logic/binary-compatibility/src/main/groovy/gradlebuild/binarycompatibility/rules/AbstractGradleViolationRule.groovy

```diff
@@ -195,4 +203,12 @@ abstract class AbstractGradleViolationRule extends AbstractContextAwareViolation
     String getCurrentVersion() {
         return context.getUserData().get("currentVersion")
     }
+
+    private String relativePathToApiChanges() {
+        if (null != apiChangesJsonFile && null != projectRootDir) {
+            return projectRootDir.relativePath(apiChangesJsonFile)
+        } else {
+            return "sample/path/to/changes.json"
```

Comment:
Seems like we never want to encounter this case. Should we throw an exception? Or better yet, perhaps remove the existing constructor which sets both of these values to null?